Main.js

```diff
@@ -1,24 +1,60 @@
-import React from 'react'
-import { StatusBar } from 'expo-status-bar'
-import { View, StyleSheet } from 'react-native'
+import React, { useState } from 'react'
+import { NavigationContainer } from '@react-navigation/native'
+import { createStackNavigator } from '@react-navigation/stack'
 
+import { upsert } from './lib/secrets'
 import { useAuthenticationContext } from './context/authentication'
 import Auth from './components/Auth'
 import Home from './components/Home'
+import TypeNewSecret from './components/TypeNewSecret'
+
+const Stack = createStackNavigator()
+
+const UI_STRINGS = {
+  routes: {
+    home: {
+      title: 'Optic',
+    },
+    type: {
+      title: 'New Secret',
+    },
+  },
+}
 
 export default function Main() {
+  const [secrets, setSecrets] = useState([])
```

Comment:
I now see what you meant with your question on slack. The secrets should not be in local state unless there's a good reason why we want to store them there. React Native has native storage that we can use, similarly to how we use localStorage in the Web app.

Reply:
I'm taking a stab at this tomorrow first thing unless you think it is better if merge this one as is and improve upon  it on a different PR. I have moved things around a little bit so it may be worth to merge as is to easy a little bit possible conflicting changes with @richiemccoll and others. Whatever y'all think is best.